### PR TITLE
Remove .dll from Interop.Libraries.HostPolicy for consistency with other runtime libraries

### LINKED ISSUE #60080: Consider removing ".dll" from the "const string HostPolicy = "hostpolicy.dll"
https://github.com/dotnet/runtime/blob/be918580c7e4ebe9b5262099caf116a67a2c9b5d/src/libraries/Common/src/Interop/Windows/Interop.Libraries.cs#L46

Notice that `hostpolicy` has extension, but `System.IO.Compression.Native` does not. The use is not all that different, so perhaps ".dll" should be removed. 

There is a very slight compat concern - it is highly unlikely, but these strings could be used as-is in some scenarios.

### PR DESCRIPTION
**Pull request by elinor-fung**
`Interop.Libraries.HostPolicy` is only used for the library name in p/invokes.

Resolves #60080

### CODE REVIEW

File: src/native/corehost/hostpolicy/hostpolicy_context.cpp [hunk 1]
```diff
@@ -53,56 +53,58 @@ namespace
...
+    const void* STDMETHODCALLTYPE pinvoke_override(const char* library_name, const char* entry_point_name)
     {
+        // This function is only called with the library name specified for a p/invoke, not any variations.
+        // It must handle exact matches to the names specified. See Interop.Libraries.cs for each platform.
 #if defined(_WIN32)
-        const char* hostPolicyLib = "hostpolicy.dll";
+        const char* hostpolicy_lib_name = "hostpolicy";
 
-        if (strcmp(libraryName, "System.IO.Compression.Native") == 0)
+        if (strcmp(library_name, "System.IO.Compression.Native") == 0)
```
Comment (jkotas) [inline src/native/corehost/hostpolicy/hostpolicy_context.cpp:63]: Should we have a macro to prepends `lib` to a string constant on Unix and use it for both System.IO.Compression.Native and hostpolicy?


```
    if (strcmp(library_name, LIBRARY_NAME("System.IO.Compression.Native")) == 0)
    {
        return CompressionResolveDllImport(entry_point_name);
    }

#if !defined(_WIN32)
    if (strcmp(library_name, LIBRARY_NAME("System.Net.Security.Native")) == 0)  
    {
....

#endif

    if (strcmp(library_name, LIBRARY_NAME("hostpolicy")) == 0)
    {
        ...
    }
```

Comment (elinor-fung) [reply src/native/corehost/hostpolicy/hostpolicy_context.cpp:63]: We have some related macros - the existing ones include the file extension and use wide/ordinary strings based on the platform. I could look at reworking and adding some.

https://github.com/dotnet/runtime/blob/85ddef115d1c279d9e1cb315b93fa218f9dfb004/src/native/corehost/hostmisc/pal.h#L76-L77

**Comment by jkotas** [comment]
Opened #76494 on the  Libraries Test Run checked coreclr windows x86 Debug crash

File: src/native/corehost/hostpolicy/hostpolicy_context.cpp [hunk 2]
```diff
@@ -53,56 +53,50 @@ namespace
...
-            if (strcmp(entrypointName, "corehost_set_error_writer") == 0)
+            if (strcmp(entry_point_name, "corehost_set_error_writer") == 0)
             {
                 return (void*)corehost_set_error_writer;
             }
         }
 
 #if defined(TARGET_OSX)
-        if (strcmp(libraryName, "libSystem.Security.Cryptography.Native.Apple") == 0)
+        if (strcmp(library_name, "libSystem.Security.Cryptography.Native.Apple") == 0)
```
Comment (jkotas) [inline src/native/corehost/hostpolicy/hostpolicy_context.cpp:97]:
```suggestion
        if (strcmp(library_name, LIB_NAME("System.Security.Cryptography.Native.Apple")) == 0)
```

File: src/native/corehost/test/nativehost/nativehost.cpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ int main(const int argc, const pal::char_t *argv[])
             }
 
             nethost_path = get_directory(nethost_path);
-            nethost_path.append(MAKE_LIBNAME("nethost"));
+            nethost_path.append(LIB_FILE_NAME_X("nethost"));
```
Comment (jkotas) [inline src/native/corehost/test/nativehost/nativehost.cpp:75]:
```suggestion
            nethost_path.append(LIB_FILE_NAME_X("nethost"));
```
For my education - why is `LIB_FILE_NAME` not sufficient here?

**Review by jkotas — APPROVED** [review]

File: src/native/corehost/test/nativehost/nativehost.cpp (hunk 3 above)
Comment (elinor-fung) [reply src/native/corehost/test/nativehost/nativehost.cpp:75]: All the host components have strings (`pal::char_t`/`pal::string_t`) as `wchar_t`/`std::wstring` for Windows and `char`/`std::string` for non-Windows internally and for exported APIs, so this uses `LIB_FILE_NAME_X` to define it according to the platform.

**Comment by elinor-fung** [comment]
Failure is https://github.com/dotnet/runtime/issues/76280.

**Pull request merged**